### PR TITLE
Rancher Prime documentation for Azure

### PR DESCRIPTION
**Pull request by guangyee**

### CODE REVIEW

File: content/rancher-prime/azure/contents.lr [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+4. Choose name for Cluster extension resource name as it suggested by the
+   tool tip "Allow alphanumeric, and dots. And the value must be 1-253
+   characters long".
+
+   <img src="./basiccluster.png" alt="Existing Resource Group" style="border: 2px solid  gray;">
+
+   ---
+   **NOTE**
+
+   The **Crete new** resource group feature is not supported.
```
Comment (mbelur) [inline content/rancher-prime/azure/contents.lr]: nit: Create

File: content/rancher-prime/azure/contents.lr [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+```
+kubectl get CspAdapterUsageRecord rancher-usage-record -o yaml
+```
+
+Output will looks like:
+
+```
+apiVersion: susecloud.net/v1
+base_product: cpe:/o:suse:rancher:v2.7.7
```
Comment (mbelur) [inline content/rancher-prime/azure/contents.lr]: base_product to show v2.7.8 since we releasing against rancher 2.7.8.

**Review by mbelur — CHANGES_REQUESTED** [review]
Looks good. Some nitpicks. The uninstall.png needs to be updated.. currently using neuvector screenshot. I guess we will cover this in a later PR

File: content/rancher-prime/azure/contents.lr (hunk 1 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr]: good catch!

File: content/rancher-prime/azure/contents.lr (hunk 2 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr]: Thanks.

**Review by chajain — APPROVED** [review]
Looks good

File: content/rancher-prime/azure/contents.lr [hunk 3]
```diff
@@ -0,0 +1,282 @@
+title: Rancher Prime in Azure: Usage Instructions
+---
+theme: rancher
+---
+body:
+
+## Prerequisites <a name="prerequisites"></a>
+
+1. A Rancher-compatible AKS cluster. Please see
+   [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-7-6/)
```
Comment (jeremy-moffitt) [inline content/rancher-prime/azure/contents.lr:10]: is there an unversioned version of this URL or will we need to update the url every time Rancher updates?

File: content/rancher-prime/azure/contents.lr [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+1. Choose the plan from the dropdown list. ( "Plans + Pricing" tab more details about Software plan)
+2. Click 'Create' 
+
+### Basics
+
+1. Select existing subscription from the dropdown list.
+2. Select existing Resource group from the dropdown list.
+3. Select existing AKS Cluster Name from the dropdown list.
+4. Choose name for Cluster extension resource name as it suggested by the
```
Comment (jeremy-moffitt) [inline content/rancher-prime/azure/contents.lr]: "as it suggested by the tool tip" reads odd... "as suggested" maybe? remove the "it"

**Review by mbelur — APPROVED** [review]

File: content/rancher-prime/azure/contents.lr [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+   ---
+   **NOTE**
+
+   The IP address of the Rancher hostname must be resolvable by a public DNS.
+
+   ---
+
+2. Using the slide bar, select the number of Rancher replicas.
+3. Choose bootstrap password as it suggested by the tip. The bootstrap
```
Comment (jeremy-moffitt) [inline content/rancher-prime/azure/contents.lr]: do you mean "as is" rather than "as it"?

File: content/rancher-prime/azure/contents.lr [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+   password will be used to authenticate to Rancher dashboard during first
```
Comment (jeremy-moffitt) [inline content/rancher-prime/azure/contents.lr]: authenticate to the Rancher dashboard

File: content/rancher-prime/azure/contents.lr [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+   login. You should change the password after the initial login.
+
+   <img src="./rancherconfiguration.png" alt="Rancher Configuration" style="border: 2px solid  gray;">
+
+click 'Next'
+
+### Review + create
+
+
+  It will summarize the offer and link to "view automation template" (Azure Resource Manager Template)
```
Comment (jeremy-moffitt) [inline content/rancher-prime/azure/contents.lr]: s/It/This/

File: content/rancher-prime/azure/contents.lr [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+
+Deployment will be in progress. After it is completed, the Rancher Prime
+Kubernetes service extension is successfully installed.
+
+ ---
+ ** NOTE: **
+
+ In the "Extensions + applications" page, the "Provisioning State" may show
+ "Succeeded" even though the provision may still be in progress. You may monitor
+ the actual progragress by loggin into the AKS cluster and follow the
```
Comment (jeremy-moffitt) [inline content/rancher-prime/azure/contents.lr]: s/loggin/logging/

File: content/rancher-prime/azure/contents.lr [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+
+## Rancher Usage Record and Billing 
+
+This command will print out CSP Adapter Usage Record:
+
+```
+kubectl get CspAdapterUsageRecord rancher-usage-record -o yaml
+```
+
+Output will looks like:
```
Comment (jeremy-moffitt) [inline content/rancher-prime/azure/contents.lr]: s/will looks/will look/

File: content/rancher-prime/azure/contents.lr [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+metadata:
+  creationTimestamp: "2023-10-03T16:45:37Z"
+  generation: 480
+  name: rancher-usage-record
+  resourceVersion: "430013"
+  uid: cf47eb21-54ae-4a35-8e75-a2362a9baa4f
+reporting_time: "2023-10-04T00:44:37Z"
+```
+
+Billing will be available in Azure Portal billing
```
Comment (jeremy-moffitt) [inline content/rancher-prime/azure/contents.lr]: available in the Azure...

File: content/rancher-prime/azure/contents.lr [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+
+Home > Cost Management <subscription> | Cost analysis
```
Comment (jeremy-moffitt) [inline content/rancher-prime/azure/contents.lr]: this <subscription> tag, does it render when running normally? in the github preview its blank

File: content/rancher-prime/azure/contents.lr [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+
+
+## Upgrading a Rancher Prime PAYG Cluster
```
Comment (jeremy-moffitt) [inline content/rancher-prime/azure/contents.lr:156]: can they also upgrade from the GUI, or is it CLI only?

Comment (jeremy-moffitt) [reply content/rancher-prime/azure/contents.lr:156]: also, can customers opt-in to automatic upgrade?

**Review by jeremy-moffitt — CHANGES_REQUESTED** [review]
Some grammar tweaks and a question about upgrade

File: content/rancher-prime/azure/contents.lr (hunk 3 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr:10]: Good question. That's the latest support matrix that is available on the Rancher doc site so we'll go with this one.

File: content/rancher-prime/azure/contents.lr (hunk 4 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr]: sound good

File: content/rancher-prime/azure/contents.lr (hunk 6 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr]: good catch!

File: content/rancher-prime/azure/contents.lr (hunk 7 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr]: Done

File: content/rancher-prime/azure/contents.lr (hunk 8 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr]: Done

File: content/rancher-prime/azure/contents.lr (hunk 10 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr]: good catch!

File: content/rancher-prime/azure/contents.lr (hunk 12 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr:156]: We can have the CLI portion in there for now. But we will update it accordingly once we have some clarity on the upgrade front.

Comment (guangyee) [reply content/rancher-prime/azure/contents.lr:156]: I am not sure if customer have the option to turn off upgrade. But that's something we can document once we have some clarity on the upgrade front.

Comment (guangyee) [reply content/rancher-prime/azure/contents.lr:156]: It does appear customer have the option to enable/disable auto upgrade. https://learn.microsoft.com/en-us/azure/virtual-machines/automatic-extension-upgrade

File: content/rancher-prime/azure/contents.lr (hunk 5 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr]: should be "as it is"

File: content/rancher-prime/azure/contents.lr (hunk 9 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr]: Done

File: content/rancher-prime/azure/contents.lr (hunk 11 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr]: I don't think this is intended to be a tag. Let me escape the brackets.

**Review by mbelur — APPROVED** [review]

**Review by jeremy-moffitt — APPROVED** [review]
Thanks for addressing my feedback! LGTM

File: content/rancher-prime/azure/contents.lr [hunk 13]
```diff
@@ -0,0 +1,280 @@
...
+```
+
+Billing will be available in the Azure Portal billing
+
+Home > Cost Management <subscription\> | Cost analysis
+
+
+## Upgrading a Rancher Prime PAYG Cluster
+
+The Azure marketplace PAYG offer is tied to a billing adapter AND Rancher Prime version. These are updated periodically as new version of the billing adapter or Rancher are released. To update the Rancher version to the latest version supported by the marketplace listing, use the "az k8s-extension update" command.
```
Comment (yarunachalam) [inline content/rancher-prime/azure/contents.lr]: I had the same content for NV, explaining offer is tied to a billing adapter .. I have received comment from Robert to re-write this content not to expose too much detail https://github.com/SUSE-Enceladus/marketplace-docs/issues/26. I have modified this for NV.

File: content/rancher-prime/azure/uninstall.png
Comment (yarunachalam) [inline content/rancher-prime/azure/uninstall.png:1]: minor issue.. type says neuvector.. May be you need to take screenshot with Rancher Plan

**Comment by yarunachalam** [comment]
@guangyee, NV Azure configuration password is exposed in the Cluster configuration. To address, we included Important Notes.. Do you see similar issue in Rancher ?? 

`The current NeuVector deployment exposes the original password in the Cluster
configuration settings in Azure Portal. Until this is resolved, it is suggested
to change the Admin password after initial login.  Edit profile in the NeuVector
console to change password.`

File: content/rancher-prime/azure/contents.lr (hunk 13 above)
Comment (yarunachalam) [reply content/rancher-prime/azure/contents.lr]: image_upg

Comment (yarunachalam) [reply content/rancher-prime/azure/contents.lr]: this is how we replaced after robert comment. Currently, I have removed this section for NV. Later, we might include upgrade section

**Review by yarunachalam — CHANGES_REQUESTED** [review]

**Comment by yarunachalam** [comment]
Thanks @guangyee for updating Password Notes. Assume, update changes will be done in a different PR once update process is finalized. I will approve this PR

**Review by yarunachalam — APPROVED** [review]

File: content/rancher-prime/azure/contents.lr (hunk 13 above)
Comment (guangyee) [reply content/rancher-prime/azure/contents.lr]: Done.

File: content/rancher-prime/azure/uninstall.png
Comment (guangyee) [reply content/rancher-prime/azure/uninstall.png:1]: That image is not being used. It's merely a placeholder for now, till we get the upgrade stuff sorted out later.

File: content/rancher-prime/azure/contents.lr [hunk 14]
```diff
@@ -0,0 +1,293 @@
...
+   for instructions on how to deploy Ingress-INGINX on AKS cluster.
+3. The Rancher hostname must be a fully qualified domain name (FQDN) and its
+   corresponding IP address must be resolvable from a public DNS. Please refer
+   to [Rancher documentation](https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/install-upgrade-on-a-kubernetes-cluster/rancher-on-aks#7-set-up-dns)
+   for instructions on how to setup DNS.
+
+##  How to install Rancher and configure Billing-adapter.
+
+A step by step walk-through for creating a new offer from the Marketplace Overview page.
+
```
Comment (rjschwei) [inline content/rancher-prime/azure/contents.lr:23]: users do not create new offers.

**Pull request merged**